pkg/build/builder/sti.go

```diff
@@ -241,8 +241,9 @@ func (s *S2IBuilder) Build() error {
 
 	if config.Incremental {
 		if s.build.Spec.Strategy.SourceStrategy.ForcePull || !isImagePresent(s.dockerClient, config.IncrementalFromTag) {
+			it, _ := dockercfg.NewHelper().GetDockerAuth(config.IncrementalFromTag, dockercfg.PullAuthType)
```

Comment:
it's counterintuitive but this should probably be dockercfg.PushAuthType

because you are pulling from the push destination, which means you would be using the same creds you are using to push to that destination.

Reply:
(it is probably working for you because the push + pull creds are the same, by default:  the SA's creds that can access the internal registry)